undertow/src/main/java/org/wildfly/extension/undertow/ListenerAdd.java

```diff
@@ -105,7 +105,9 @@ public HttpHandler wrap(HttpHandler handler) {
         serviceBuilder.addDependency(workerServiceName, XnioWorker.class, service.getWorker())
                 .addDependency(socketBindingServiceName, SocketBinding.class, service.getBinding())
                 .addDependency(bufferPoolServiceName, (Injector) service.getBufferPool())
-                .addDependency(UndertowService.SERVER.append(serverName), Server.class, service.getServerService());
+                .addDependency(UndertowService.SERVER.append(serverName), Server.class, service.getServerService())
+                // Include the capability-declared service name as an alias  TODO get rid of the old style name
+                .addAliases(ListenerResourceDefinition.LISTENER_CAPABILITY.getCapabilityServiceName(name));
```

Comment:
nevermind my comment, I get it why you used alias.
I should probably push my branch that has more than just this changes on it. 
This is why i am confusing what is "done" and what is done.